### ISSUE TITLE
One error and tons of warning on PPC64le

### ISSUE THREAD

**Issue by kif** [silx-kit/silx#2029]
```
WARNING:py.warnings:/pwrwork/rech/gjy/uwk35yz/anaconda3/lib/python3.6/importlib/_bootstrap_external.py:426: ImportWarning: Not importing directory /pwrwork/rech/gjy/uwk35yz/anaconda3/lib/python3.6/site-packages/mpl_toolkits: missing __init__
  _warnings.warn(msg.format(portions[0]), ImportWarning)

WARNING:py.warnings:/pwrwork/rech/gjy/uwk35yz/anaconda3/lib/python3.6/importlib/_bootstrap_external.py:426: ImportWarning: Not importing directory /gpfs16l/pwrwork/rech/gjy/uwk35yz/anaconda3/lib/python3.6/site-packages/mpl_toolkits: missing __init__
  _warnings.warn(msg.format(portions[0]), ImportWarning)

WARNING:run_tests:Test silx 0.9.0-dev0 from /gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx
WARNING:silx.test:silx.sx tests disabled Skipped by command line
WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/utils.py:86: DeprecationWarning: invalid escape sequence \*
  """

WARNING:py.warnings:/pwrwork/rech/gjy/uwk35yz/anaconda3/lib/python3.6/importlib/_bootstrap.py:219: ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
  return f(*args, **kwds)

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/configdict.py:150: DeprecationWarning: invalid escape sequence \,
  sstr = sstr.replace("\,", ",").replace("^@", ",")

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/configdict.py:194: DeprecationWarning: invalid escape sequence \,
  if sstr.count(",") == sstr.count("\,"):

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/configdict.py:216: DeprecationWarning: invalid escape sequence \,
  sstr.replace("\,", "^@")

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/configdict.py:263: DeprecationWarning: invalid escape sequence \,
  """

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/configdict.py:264: DeprecationWarning: invalid escape sequence \,
  return str(self.replace("\,", ",").lstrip("\\"))

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/configdict.py:320: DeprecationWarning: invalid escape sequence \,
  """

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/configdict.py:471: DeprecationWarning: invalid escape sequence \`
  """

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/configdict.py:476: DeprecationWarning: invalid escape sequence \,
  sstr = sstr.replace(",", "\,")

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/spech5.py:365: DeprecationWarning: invalid escape sequence \d
  year_rx = '(?P<year>\d{4})'

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/spech5.py:366: DeprecationWarning: invalid escape sequence \d
  day_nb_rx = '(?P<day_nb>[0-3 ]\d)'

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/spech5.py:367: DeprecationWarning: invalid escape sequence \d
  month_nb_rx = '(?P<month_nb>[0-1]\d)'

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/spech5.py:368: DeprecationWarning: invalid escape sequence \d
  hh_rx = '(?P<hh>[0-2]\d)'

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/spech5.py:369: DeprecationWarning: invalid escape sequence \d
  mm_rx = '(?P<mm>[0-5]\d)'

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/spech5.py:370: DeprecationWarning: invalid escape sequence \d
  ss_rx = '(?P<ss>[0-5]\d)'

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/spech5.py:371: DeprecationWarning: invalid escape sequence \d
  tz_rx = '(?P<tz>[+-]\d\d:\d\d){0,1}'

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/utils/number.py:56: DeprecationWarning: invalid escape sequence \s
  _parse_numeric_value = re.compile("^\s*[-+]?0*(\d+?)?(?:\.(\d+))?(?:[eE]([-+]?\d+))?\s*$")

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/io/test/test_utils.py:74: DeprecationWarning: invalid escape sequence \.
  """

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/math/fit/fittheory.py:87: DeprecationWarning: invalid escape sequence \*
  """

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/math/fit/fittheory.py:153: DeprecationWarning: invalid escape sequence \*
  instead of the usual fit function signature."""

WARNING:py.warnings:/pwrwork/rech/gjy/uwk35yz/anaconda3/lib/python3.6/importlib/_bootstrap.py:219: ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
  return f(*args, **kwds)

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/math/calibration.py:56: DeprecationWarning: invalid escape sequence \m
  """

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/math/calibration.py:66: DeprecationWarning: invalid escape sequence \m
  """

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/math/calibration.py:86: DeprecationWarning: invalid escape sequence \m
  """

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/math/calibration.py:111: DeprecationWarning: invalid escape sequence \m
  :param x1: Calibration array"""

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/math/calibration.py:156: DeprecationWarning: invalid escape sequence \m
  :param function: Calibration function"""

WARNING:silx.opencl.common:Unable to import pyOpenCl. Please install it from: http://pypi.python.org/pypi/pyopencl
WARNING:silx.opencl.codec.byte_offset:No PyOpenCL, no byte-offset, please see fabio
WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/app/convert.py:61: DeprecationWarning: invalid escape sequence \d
  """

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/app/convert.py:64: DeprecationWarning: invalid escape sequence \.
  pattern_string = pattern_string.replace(".", "\.")

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/app/convert.py:67: DeprecationWarning: invalid escape sequence \d
  pattern_string = pattern_string.replace("%d", "([-+]?\d+)")

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/app/convert.py:70: DeprecationWarning: invalid escape sequence \d
  for sub_pattern in re.findall("%0\d+d", pattern_string):

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/app/convert.py:71: DeprecationWarning: invalid escape sequence \d
  n = int(re.search("%0(\d+)d", sub_pattern).group(1))

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/app/convert.py:73: DeprecationWarning: invalid escape sequence \d
  re_sub_pattern = "([+-]?\d)"

WARNING:py.warnings:/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/app/convert.py:75: DeprecationWarning: invalid escape sequence \d
  re_sub_pattern = "([\d+-]\d{%d})" % (n - 1)

WARNING:silx.gui.test:silx.gui tests disabled: Skipped by command line
WARNING:silx.opencl.sift.match:No PyOpenCL, no sift
WARNING:silx.opencl.sift.alignment:No PyOpenCL, no sift
WARNING:silx.opencl.sift.test.test_matching:feature module is not available to compare results with C++ implementation. Matching cannot be tested.
ss................................................................................................................................E..................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................s......................................sssssssssssssssssssssssssssssssssssssssssssssssss.....
======================================================================
ERROR: testLosePrecisionUsingFloat80 (silx.utils.test.test_number.TestConversionTypes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/utils/test/test_number.py", line 154, in testLosePrecisionUsingFloat80
    dtype = func(value)
  File "/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/utils/testutils.py", line 227, in wrapper
    result = func(*args, **kwargs)
  File "/gpfs16l/pwrhome/rech/grpgen/uwk35yz/workspace/silx/build/lib.linux-ppc64le-3.6/silx/utils/testutils.py", line 183, in __exit__
    expected_count, logging.getLevelName(level), count))
RuntimeError: Expected 1 WARNING logging messages, got 0
```

**Comment by vasole** [comment]
Could the regular expressions warnings be prevented by prepending the letter r to the strings?

**Comment by alexmarie78** [comment]
Hello,
We got the same error on arm64 [0]  with silx 0.10.1 : 

```
ERROR: testLosePrecisionUsingFloat80 (silx.utils.test.test_number.TestConversionTypes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/.pybuild/cpython2_2.7_silx/build/silx/utils/test/test_number.py", line 154, in testLosePrecisionUsingFloat80
    dtype = func(value)
  File "/<<PKGBUILDDIR>>/.pybuild/cpython2_2.7_silx/build/silx/utils/testutils.py", line 227, in wrapper
    result = func(*args, **kwargs)
  File "/<<PKGBUILDDIR>>/.pybuild/cpython2_2.7_silx/build/silx/utils/testutils.py", line 183, in __exit__
    expected_count, logging.getLevelName(level), count))
RuntimeError: Expected 1 WARNING logging messages, got 0
```
The error message is terse, so I don't know how to debug this one.
Hope you can help me.
Cheers,
Alex

[0] https://buildd.debian.org/status/fetch.php?pkg=silx&arch=arm64&ver=0.10.1%2Bdfsg-1%7Eexp1&stamp=1559037411&raw=0

**Comment by picca** [comment]
So it seems that on arm64 the skipFloat80 is not triggerd
Now a warning should  be generated during the test. but this is not true  on the arm64.
Maybe it is normal.

can you be more specific on which warning should be triggered ?

**Comment by picca** [comment]
We instrumented the min_numerical_convertible_type method and got this:

```
('string : ', '1000000000.00001013332', 'check_accuracy : ', True, 'numpy_type : ', <type 'numpy.float128'>, 'biggest_float : ', <type 'numpy.float128'>)
```

**Comment by picca** [comment]
the result is there.
result: 1000000000000010133320e+09

so there is no accuray problem on arm64.
Maybe the problem is  related to the sjkip part.

**Comment by picca** [comment]
Someone from the ERSF can tel us what is the right fix and if our analyse is right.
then we will provide a patch.

Cheers

**Comment by picca** [comment]
here the test:

```
    def skipIfFloat80NotSupported(self):
        if number.is_longdouble_64bits():
            self.skipTest("float-80bits not supported")

    def testLosePrecisionUsingFloat80(self):
        self.skipIfFloat80NotSupported()
        if pkg_resources.parse_version(numpy.version.version) <= pkg_resources.parse_version("1.10.4"):
            self.skipTest("numpy > 1.10.4 expected")
        value = "1000000000.00001013332"
        func = testutils.test_logging(number._logger.name, warning=1)
        func = func(number.min_numerical_convertible_type)
        dtype = func(value)
        self.assertIn(dtype, (numpy.longdouble, ))
```

So we skip the test (if number.is_longdouble_64bits()) if longdoulbe == 64bits.
In our case we have 128 bits, so the test is not skipped, and numpy does not complain about precision because we have 128 bits.
What is the inetnded expectations of thei test ? (arch, size of the  longdouble etc...)

**Comment by t20100** [comment]
Hi,

This is fixed in silx v0.11
Corresponding PR is #2626
The test checks that a warning is issued when the conversion to the return type will lead to a loss of precision.

The issue comes from the test initially expecting maximum mantissa to be that of `long double` on x86 (i.e., float stored on 80 bits), while other architecture are providing true float128 (ppc64le and arm64 for instance).

**Comment by t20100** [comment]
Closing as warnings (PR #2030) and error (PR #2626) are fixed in current master (and v0.11).
